### DISCUSSION TITLE
Make gitversion start with 1.0.0 instead of 0.1.0

### DISCUSSION THREAD

**Discussion by bitbonk** [GitTools/GitVersion#3123] [Q&A] [answered]
I have a fresh new git repository that has not seen any releases and so far only has some commits on branch `develop`. I have a default (completely empty) `GitVersion.yml` so I am planning to use the default behavior with the default suggested branch names.

Currently the nightly builds of branch `develop` produce Nuget packages with the following version: `My.Package.0.1.0-alpha0005`.

What is the best way to make gitversion start with the initial version of `1.0.0` - so that it works well for all branches:

e.g.
- branch `develop` should produce `My.Package.1.0.0-alpha0000`
- branch `release/1.0.0` should produce `My.Package.1.0.0-beta0000`

**Comment by gep13** [comment, marked as answer]
The suggested approach would be to use the `next-version` attribute in the GitVersion.yml file.  This is mentioned in the docs on this page:

https://gitversion.net/docs/reference/version-increments

As well as the Configuration section:

https://gitversion.net/docs/reference/configuration

So you would add:

```yaml
next-version: 1.0.0
```

In the yaml file, and GitVersion would begin asserting the version number for the repository, using this as the starting point.

**Reply by bitbonk** [reply]
@gep13 Tanks, and when could I remove it from the yaml? After 1.0.0 was released? (I plan to trigger a release build after tagging a commit on main with `1.0.0`.)

**Reply by gep13** [reply]
After the repository is tagged, the version number from the tag takes precedence over the version number from the GitVersion.yml file.  So yes, once tagged, you can remove the next-version from the yml file.  Then when you get ready to work towards a 2.0.0 release, you can add it back in to force that version number to be used as the base version for the asserted version numbers.
